### ISSUE TITLE
Crash on resizing artwork (ffmpeg?)

### ISSUE THREAD

**Issue by gitteric** [ejurgensen/forked-daapd#137]
I discovered an issue when forked-daapd tries to scale an Artwork where the embedded jpg is somehow corrupt. Shure I'll now vaildate my files for correct jpgs. But the deamon shouldn't break in case of bad artwork, i guess. The log tells it's related to an ffmpeg call for resizing. Maybe I just use an outdatet library (libav?). Fix or help is appreciated :)

Heres the tail of the log, "artwork: Artwork too large, rescaling image" is the last entry:

[2015-04-23 19:46:16] [DEBUG]   ffmpeg: marker=ee avail_size_in_buf=24328
[2015-04-23 19:46:16] [DEBUG]   ffmpeg: marker parser used 14 bytes (112 bits)
[2015-04-23 19:46:16] [DEBUG]   ffmpeg: marker=db avail_size_in_buf=24312
[2015-04-23 19:46:16] [DEBUG]   ffmpeg: index=0
[2015-04-23 19:46:16] [DEBUG]   ffmpeg: qscale[0]: 1
[2015-04-23 19:46:16] [DEBUG]   ffmpeg: index=1
[2015-04-23 19:46:16] [DEBUG]   ffmpeg: qscale[1]: 5
[2015-04-23 19:46:16] [DEBUG]   ffmpeg: marker parser used 132 bytes (1056 bits)
[2015-04-23 19:46:16] [DEBUG]   ffmpeg: marker=c0 avail_size_in_buf=24178
[2015-04-23 19:46:16] [DEBUG]   ffmpeg: sof0: picture: 500x450
[2015-04-23 19:46:16] [DEBUG]   ffmpeg: component 0 1:1 id: 0 quant:0
[2015-04-23 19:46:16] [DEBUG]   ffmpeg: component 1 1:1 id: 1 quant:1
[2015-04-23 19:46:16] [DEBUG]   ffmpeg: component 2 1:1 id: 2 quant:1
[2015-04-23 19:46:16] [DEBUG]   ffmpeg: component 3 1:1 id: 3 quant:0
[2015-04-23 19:46:16] [DEBUG]   ffmpeg: pix fmt id 11111111
[2015-04-23 19:46:16] [ WARN]   ffmpeg: Unhandled pixel format 0x11111111
[2015-04-23 19:46:16] [ WARN]   ffmpeg: max_analyze_duration reached
[2015-04-23 19:46:16] [ WARN]   ffmpeg: Estimating duration from bitrate, this m                                        ay be inaccurate
[2015-04-23 19:46:16] [DEBUG]  artwork: Found embedded artwork in '/mnt/sdb1/Mp3                                        _Tagged/Electronic/Schiller/Atemlos/CD1/01 - Schiller - Willkommen.mp3'
[2015-04-23 19:46:16] [DEBUG]  artwork: Original image dimensions: w 500 h 450
[2015-04-23 19:46:16] [DEBUG]  artwork: Raw destination width 53 height 47
[2015-04-23 19:46:16] [DEBUG]  artwork: Destination width 52 height 47
[2015-04-23 19:46:16] [DEBUG]  artwork: Artwork too large, rescaling image

**Comment by ejurgensen** [comment]
Yes, the rescaling is done by ffmpeg or libav. Which version do you have? Does it also crash if you try rescaling with the command-line tool? If it does, there is not so much I can do. It would then be good if you filed the bug with your distro.

If you built forked-daapd yourself, and you don't want to upgrade ffmpeg/libav, a quick workaround is to insert "return 0" at the beginning of rescale_needed() in artwork.c. Many clients don't actually require the server to rescale.

**Comment by gitteric** [comment]
Thanks for your fast reply. Im running latest libav library available from ubuntu repos (~9.18).
It tried rescaling the corrupt jpg with older ffmpeg libs on my windows machine (which crashes with same message) and with latest ffmpeg executable on my windows machine (which gave a warning but worked).
I thought it might be relatet to a false pixel encoding detection. So I tried to add a check of dst->pix_fmt for that "Unhandled pixel format" in artwork.c/artwork_rescale... But as I had not much idea of what I was doing - it just failed. Maybe I'll give it a try after I gained some real C skills ;)

However, disabling rescale like you suggested worked out pretty well. For now I'm fine with that. 
 
Let me know if there are any news on that or forked-daapd in general.
Thanks & best regs
Eric
